src/main/java/semantics/RDFDatasetDirectStatementDeleter.java

```diff
@@ -0,0 +1,349 @@
+                st.getContext() != null ? st.getContext().stringValue() : null,
+                params);
+            Result result = graphdb.execute(cypher, params);
+            if (result.hasNext()) {
+              node = (Node) result.next().get("n");
+              if (result.hasNext()) {
+                String props =
+                    "{uri: " + st.getObject().stringValue() +
+                        (st.getContext() == null ? "}" :
+                            ", graphUri: " + st.getContext().stringValue() + "}");
+                throw new IllegalStateException(
+                    "There are multiple matching nodes for the given properties " + props);
+              }
+            } else {
+              throw new NoSuchElementException(
+                  "There exists no node with \"uri\": " + st.getSubject().stringValue()
+                      + " and \"graphUri\": " + st.getContext().stringValue());
+            }
+            return node;
+          }
+        });
+      } catch (InvalidCacheLoadException | NoSuchElementException e) {
+        e.printStackTrace();
+      }
+      if (toNode == null) {
+        notDeletedStatementCount++;
+        continue;
+      }
+
+      // find relationship if it exists
+      if (fromNode.getDegree(RelationshipType.withName(handleIRI(st.getPredicate(), RELATIONSHIP)),
+          Direction.OUTGOING) <
+          toNode.getDegree(RelationshipType.withName(handleIRI(st.getPredicate(), RELATIONSHIP)),
+              Direction.INCOMING)) {
+        for (Relationship rel : fromNode
+            .getRelationships(RelationshipType.withName(handleIRI(st.getPredicate(), RELATIONSHIP)),
+                Direction.OUTGOING)) {
+          if (rel.getEndNode().equals(toNode)) {
+            rel.delete();
+            break;
+          }
+        }
+      } else {
+        for (Relationship rel : toNode
+            .getRelationships(RelationshipType.withName(handleIRI(st.getPredicate(), RELATIONSHIP)),
+                Direction.INCOMING)) {
+          if (rel.getStartNode().equals(fromNode)) {
+            rel.delete();
+            break;
+          }
+        }
+      }
+      deleteNodeIfEmpty(toNode);
+      deleteNodeIfEmpty(fromNode);
+    }
+
+    statements.clear();
+    resourceLabels.clear();
+    resourceProps.clear();
+    if (notDeletedStatementCount > 0) {
+      setbNodeInfo(bNodeCount
+          + " of the statements could not be deleted, due to containing a blank node.");
+    }
+
+    //TODO what to return here? number of nodes and rels?
+    return 0;
+  }
+
+  @Override
+  protected void periodicOperation() {
+    Util.inTx(graphdb, this);
+    totalTriplesMapped += mappedTripleCounter;
+    log.info("Successful partial commit of " + mappedTripleCounter + " triples. " +
+        (totalTriplesMapped - notDeletedStatementCount) + " triples deleted so far...");
+    mappedTripleCounter = 0;
+  }
+
+  Map<String, String> getNamespaces() {
```

Comment:
Probably a case of bad design: this parameter belongs to the super-class, so the super-class should provide a getter for it.